src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -327,8 +327,9 @@ static NPyMechObj* new_pymechobj(NPySegObj* pyseg, Prop* p) {
     if (!m) {
         return NULL;
     }
+    Py_INCREF(pyseg);
+    Py_XDECREF(m->pyseg_);
```

Comment:
how is it known that `m->pyseg_` is not null?

Reply:
Sigh, always get the `X` version backwards.
Is the `m->pyseg` guaranteed to be initialized `null`?
I see it is `PyObject_New`d in `new_pymechobj`, and the doc for `PyObject_New` say that `Fields not defined by the Python object header are not initialized.` (so I'm not sure if they object is bzero`d or not?

Reply:
I'm set to discover that... maybe that's the reason of the cur seg faults :(